src/plugins/data_source/server/util/decide_client.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { RequestHandlerContext } from 'src/core/server';
+
+export const decideLegacyClient = (
+  requestContext: RequestHandlerContext,
+  dataSourceId: string | null
```

Comment:
nit: let's use optional parameter instead of null.

Reply:
We have decideClient in https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/server/search/opensearch_search/decide_client.ts.

1. Can we use same file for `decideLegacyClient` ?
2. Can we use request to get dataSourceId similar to decideClient? 
OR
maybe refactor decideClient to return legacy client base on additional parameter?

3. I can see in decideClient we are supporting LongNumerals. Is it something we need to take for legacy client? Can you double check?

Reply:
I found this function https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b4130aa2c1c80033d1bec4d18d44e20612ac0784/src/plugins/data/server/index_patterns/routes.ts#L158 for legacy client.

Reply:
Wrote a workaround to use `decideClient()` but this will need a refactor because other plugins may need this function as well. Raised a new issue to address this #6417